File: derby/java/update/derby_update.java
// ----------------------------------------------------------------------
/*
	java/update/derby_update.java

					Oct/01/2012

*/
// ----------------------------------------------------------------------
import java.util.Date;
import java.sql.SQLException;
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.Statement;
import java.util.Properties;

// ----------------------------------------------------------------------
public class derby_update
{

// ----------------------------------------------------------------------
public static void main (String[] args)
	throws SQLException, ClassNotFoundException,
	InstantiationException, IllegalAccessException
{
	System.out.println ("*** 開始 ***");

	String	id = args[0];
	int	population = Integer.parseInt (args[1]);
	System.out.print ("\tid = " + id);
	System.out.println ("\tpopulation = " + population);

	String str_connect= "jdbc:derby://localhost:1527/city_aaa;create=true";
	Connection conn = DriverManager.getConnection (str_connect);

	System.out.println ("*** connected. ***");

	rdb_common.update_proc	(conn,id,population);

	rdb_common.display_proc	(conn);

	conn.close ();

	System.out.println ("*** 終了 ***");
}

// ----------------------------------------------------------------------
}
// ----------------------------------------------------------------------

File: derby/scala/update/derby_update.scala
// -----------------------------------------------------------------
//	update/derby_update.scala
//
//						Aug/01/2011
// -----------------------------------------------------------------
import scala.io.Source
import java.sql._

// -----------------------------------------------------------------
object derby_update
{
// -----------------------------------------------------------------
def main (args: scala.Array[String])
{
	println	("*** 開始 ***")

	val id_in = args(0)
	val population_in = Integer.parseInt (args(1))

	println (id_in + "\t" + population_in)

	update_proc (id_in,population_in)

	println	("*** 終了 ***")
}

// -----------------------------------------------------------------
def update_proc (id_in:String,population_in:Int)
{
	val str_connect= "jdbc:derby://localhost:1527/city_aaa;create=true"
	val conn = DriverManager.getConnection (str_connect)

	sql_manipulate.update_proc (conn,id_in,population_in)
	sql_manipulate.display_proc (conn)
	conn.close
}

// -----------------------------------------------------------------
}
// -----------------------------------------------------------------
